src/commands/new.rs

```diff
@@ -28,21 +44,39 @@ impl NewCommand {
         self.run_template(repo, id).await
     }
 
+    async fn list_templates(&self) -> anyhow::Result<()> {
+        let entries = hub_api::index().await.unwrap();
+        let matches = entries.iter()
+            .filter(|e| self.is_terms_match(e))
+            .sorted_by_key(|e| e.title())
+            .collect_vec();
+
+        if matches.is_empty() {
+            println!("No templates match your search terms");
+            return Ok(());
+        }
+
+        for entry in matches {
+            println!("Template: {}\nDescription: {}\n", entry.title(), entry.summary());
```

Comment:
(if we keep this)

Rather than printing the ID and description on alternate lines, consider using a table.  You can see an example of this at https://github.com/itowlson/spin-hub-cli/blob/986b946b7cc15965c03fd6b24f5bb9d313d1a751/src/commands/search.rs#L32